.github/workflows/push-aptos-node.yaml

```diff
@@ -19,18 +37,27 @@ jobs:
         username: '${{ secrets.DOCKERHUB_USERNAME }}'
     - uses: 'docker/build-push-action@v6'
       with:
+        build-args: |
+          PUBLISHER_PK=${{ env.PUBLISHER_PK }}
+          BIG_MONEY_GUY_PK=${{ env.BIG_MONEY_GUY_PK }}
         cache-from: 'type=gha'
         cache-to: 'type=gha,mode=max'
         context: '.'
-        file: 'src/docker/aptos-node/Dockerfile'
+        file: 'src/docker/emojicoin-dot-fun-aptos-node/Dockerfile'
         labels: '${{ steps.metadata.outputs.labels }}'
-        platforms: '${{ vars.DOCKER_IMAGE_PLATFORMS }}'
+        # The emojicoin-node image can only be built for the platform on which
+        # the build is running. This is because the build process requires
+        # running `aptos node run-localnet`, which spawns a new container. This
+        # new container must be able to run the `aptos` binary, which is only
+        # available for the platform on which the build is running.
```

Comment:
Functionally I totally understand that in practice this just wasn't working on GitHub actions. Theoretically, as to

> only available for the platform on which the build is running

I suspect that something could be done about this unless you can point me to a reference in docs that explicitly say "you can't do x", but definitely not necessary right now

Because as I understand it, basically everything is cached except maybe the state generation, which someone can always run once if they need to do local testing, and which should take less than a minute

Feel free to resolve this comment unless there is something you disagree on

Reply:
I may not have written that as clear as I could have; it's not so much "you can't do this in Docker" but more that we'd have to find a solution for getting the container created from `aptos node run-localnet` to match the architecture of the host machine rather than the Docker image that runs the `aptos node run-localnet` command.

The docker startup script in the `run-localnet` command pulls the image at runtime, I assume based on the `DOCKER_DEFAULT_PLATFORM` env var, see [here](https://github.com/aptos-labs/aptos-core/blob/f39917abb975872c447c4aeb4001e9c0248ec13d/crates/aptos/src/node/local_testnet/indexer_api.rs#L108) where Daniel warns about setting it.

Since `Illegal instruction` is explicitly the error code for an incompatible opcode based on the running machine's processor architecture and that's the error I was running into, I assume that's the reason the CLI was not really intended to be run inside a container.

We could:
  1. apply a patch to the CLI docker startup scripts
  2. Try using `DOCKER_DEFAULT_PLATFORM`

But I actually am not sure we should have (or even need) pre-generated state inside a container anymore, for a few reasons that we can discuss later on a call (it will take a bit to explain). The jist of it though is that it's more difficult than I realized and I'm not sure it makes sense anyway

In terms of orchestration we can always just wait on the readiness endpoint aka `http://localhost:8070` to be up, and then go from there.

The main problem that we were running into was more of a devex thing where we weren't properly cleaning up the containers in relation to our indexer processors, but it doesn't need docker orchestration to achieve that- it just needs to be approached holistically instead of piecemeal

Reply:
> I actually am not sure we should have (or even need) pre-generated state inside a container anymore

Agreed, it's only a few commands to publish and fund, and the amount of time it takes is negligible compared to how long it takes the site to boot/run other e2e commands

Probably cleanest to not have anything extra in it

Feel free to mark as resolved